packages/manager/apps/pci-kubernetes/src/pages/new/steps/NodePoolStep.component.tsx

```diff
@@ -101,71 +128,117 @@ const NodePoolStep = ({
-              onFlavorChange={setFlavor}
-            />
-          </div>
-          <div className="mb-8">
-            <NodePoolSize
-              isMonthlyBilled={isMonthlyBilled}
-              onScaleChange={setScaling}
-              antiAffinity={antiAffinity}
-            />
-          </div>
-          <div className="mb-8">
-            <NodePoolAntiAffinity
-              isChecked={antiAffinity}
-              isEnabled={!scaling?.isAutoscale}
-              onChange={setIsAntiaffinity}
-            />
-          </div>
-          <div className="mb-8">
-            <BillingStep
-              price={price.hour}
-              monthlyPrice={price.month}
-              monthlyBilling={{
-                isComingSoon: isPricingComingSoon,
-                isChecked: isMonthlyBilled,
-                check: setIsMonthlyBilled,
-              }}
-              warn={scaling?.isAutoscale && isMonthlyBilled}
+      <div className="bo border-">
+        {!stepper.node.step.isLocked && nodePoolEnabled && (
+          <>
+            <div className="mb-8">
+              <NodePoolName
+                onTouched={setIsTouched}
+                hasError={hasError}
+                onNameChange={setName}
+                name={name}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolType
+                projectId={projectId as string}
+                region={stepper.form.region.name}
+                onFlavorChange={setFlavor}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolSize
+                isMonthlyBilled={isMonthlyBilled}
+                onScaleChange={setScaling}
+                antiAffinity={antiAffinity}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolAntiAffinity
+                isChecked={antiAffinity}
+                isEnabled={!scaling?.isAutoscale}
+                onChange={setIsAntiaffinity}
+              />
+            </div>
+            <div className="mb-8">
+              <BillingStep
+                price={price.hour}
+                monthlyPrice={price.month}
+                monthlyBilling={{
+                  isComingSoon: isPricingComingSoon ?? false,
+                  isChecked: isMonthlyBilled,
+                  check: setIsMonthlyBilled,
+                }}
+                warn={(scaling?.isAutoscale && isMonthlyBilled) ?? false}
+              />
+            </div>
+          </>
+        )}
+        {!stepper.node.step.isLocked && (
+          <OsdsButton
+            variant={ODS_BUTTON_VARIANT.stroked}
+            className="my-6 w-fit"
+            disabled={!nodes || !scaling || !flavor || undefined}
+            size={ODS_BUTTON_SIZE.sm}
+            color={ODS_TEXT_COLOR_INTENT.primary}
+            {...(isButtonDisabled ? { disabled: true } : {})}
+            onClick={() => {
+              if (nodes || scaling || flavor) {
```

Comment:
I think this condition should be the same as the isButtonDisabled, or is there a case where the button shouldn't be disabled but shouldn't do any action ?